Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -63,7 +144,23 @@ public static int[] RemoveAt(int[] array, int index)
         /// <returns>A new array with element added at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertFirst(int[] array, int number)
         {
-            // ToDo: implement.
+            if (array != null && array.Length != 0)
+            {
+                var arrayLength = array.Length;
+                int[] newArray = new int[arrayLength + 1];
+                // insert first element manually and copy existing elements
+                newArray[0] = number;
+                for (var i = 0; i < arrayLength; i++)
+                {
+                    newArray[i+1] = array[i];
+                }
+                array = newArray;
+            }
+            else
+            {
+                var newArray = new[] { number };
+                array = newArray;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'array'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5011611)